external-crates/move/crates/move-package/src/compilation/compiled_package.rs

```diff
@@ -756,3 +765,123 @@ pub(crate) fn make_source_and_deps_for_compiler(
     };
     Ok((source_package_paths, deps_package_paths))
 }
+
+/// partitions `deps` by whether we need to compile dependent packages with a
```

Comment:
Can you explain if we handle this case?

- `A` requires `B` at version `0.1.0`
- `A` requires `C` at version `0.2.0`
- `B` requires `C` at version `0.1.0`

If not, I do think we need a map of versions to modules to compile, so any individual module can find its dependencies at the right version.

Reply:
Based on how we currently resolve dependencies, I'm not sure we can / want to / need to support this case. 

If packages in the dependency tree require `C` at different versions (more plainly, if different `C` dependencies are specified in `Move.toml`s based on package name) [we detect a conflict](https://github.com/MystenLabs/sui/blob/7f6278d88e690306c968a4515bd1f13d1780aca9/external-crates/move/crates/move-package/src/resolution/dependency_graph.rs#L630-L631). So my understanding based on how we do resolution (which is not entirely complete :P) is that if `C` exists in the dependency graph it can only be one version.